test/e2e/reconciler/scheduling/upsynced_scheduling_test.go

```diff
@@ -0,0 +1,197 @@
+		metav1.GetOptions{},
+	)
+	require.NoError(t, err)
+
+	t.Log(t, "Wait for being able to list deployments in the consumer workspace via direct access")
+	require.Eventually(t, func() bool {
+		_, err := upstreamKubeClusterClient.Cluster(userWsPath).CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+		return !apierrors.IsNotFound(err)
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	stateLabelKey := "state.workload.kcp.io/" + workloadv1alpha1.ToSyncTargetKey(synctargetWsName, syncTarget.Name)
+
+	t.Log("Upsyncing Pod to KCP")
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: upstreamNamespace.Name,
+			Labels: map[string]string{
+				stateLabelKey: "Upsync",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "test-container",
+				},
+			},
+		},
+	}
+
+	// Create a client that uses the upsyncer URL
+	upsyncerKCPClient, err := kcpkubernetesclientset.NewForConfig(syncerFixture.UpsyncerVirtualWorkspaceConfig)
+	require.NoError(t, err)
+
+	_, err = upsyncerKCPClient.Cluster(userWsName.Path()).CoreV1().Pods(upstreamNamespace.Name).Create(ctx, &pod, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Log("Checking that the upsynced POD has the state set to Upsync...")
+	framework.Eventually(t, func() (bool, string) {
+		_, err := upstreamKubeClusterClient.Cluster(userWsPath).CoreV1().Pods(upstreamNamespace.Name).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+		if pod.Labels[stateLabelKey] == "Upsync" {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected state to be Upsync, got %s", pod.Labels[stateLabelKey])
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "expected state to be Upsync, got %s", pod.Labels[stateLabelKey])
+
+	t.Log("Stopping the syncer healthchecker...")
+	syncerFixture.StopHeartBeat(t)
+
+	t.Log("Checking that the synctarget is not ready...")
+	framework.Eventually(t, func() (bool, string) {
+		syncTarget, err := upstreamKcpClient.Cluster(synctargetWsPath).WorkloadV1alpha1().SyncTargets().Get(ctx, syncTarget.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+		if conditions.IsTrue(syncTarget, workloadv1alpha1.HeartbeatHealthy) {
+			return false, "expected synctarget to be not ready"
+		}
+		return true, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Log("Checking that the upsynced POD remains in the Upsync state...")
+	require.Never(t, func() bool {
+		_, err := upstreamKubeClusterClient.Cluster(userWsPath).CoreV1().Pods(upstreamNamespace.Name).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		return pod.Labels[stateLabelKey] != "Upsync"
+	}, 5*time.Second, time.Millisecond*100, "expected state to be Upsync, got %s", pod.Labels[stateLabelKey])
+
+	t.Log("Starting the syncer healthcheck again...")
+	syncerFixture.StartHeartBeat(t)
+	syncerFixture.WaitForSyncTargetReady(ctx, t)
```

Comment:
Afaict this is already included in `StartHeartBeat()`

Reply:
done.